### PR TITLE
#1363

### PR DESCRIPTION
**Pull request by dot5enko**
Created flexible component which uses simple templates to generate refill links.

### CODE REVIEW

File: app/components/Utility/CryptoLinkFormatter.jsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    };
+
+    static defaultProps = {
+        size: 140
+    };
+
+    constructor(params) {
+        super(params);
+
+        this.assetTemplates = {
```
Comment (sschiessl-bcp) [inline app/components/Utility/CryptoLinkFormatter.jsx]: Please document how assetTemplates dictionary needs to look like, including bind and value usage

File: app/components/Utility/CryptoLinkFormatter.jsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+ *  Given an asset name, amount and label generates uri for its refilling
+ *
+ */
+
+class CryptoLinkFormatter extends React.Component {
+    static assetTemplates = {};
+
+    static propTypes = {
+        asset: PropTypes.string,
+        address: PropTypes.string,
```
Comment (sschiessl-bcp) [inline app/components/Utility/CryptoLinkFormatter.jsx]: Asset and address are mandatory parameters

File: app/components/Utility/CryptoLinkFormatter.jsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        amount: PropTypes.number,
+        message: PropTypes.string,
+        size: PropTypes.number
+    };
+
+    static defaultProps = {
```
Comment (sschiessl-bcp) [inline app/components/Utility/CryptoLinkFormatter.jsx]: Please include all non-mandatory props in defaults

File: app/components/Utility/CryptoLinkFormatter.jsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            tokenName
+        ) {
+            if (tokenName in conf) {
+                return conf[tokenName];
+            } else {
+                return true;
+            }
+        });
+
+        if (error) {
```
Comment (sschiessl-bcp) [inline app/components/Utility/CryptoLinkFormatter.jsx]: error is always false?

File: app/components/Utility/CryptoLinkFormatter.jsx (hunk 1 above)
Comment (dot5enko) [reply app/components/Utility/CryptoLinkFormatter.jsx]: stressed this info in comments

File: app/components/Utility/CryptoLinkFormatter.jsx (hunk 2 above)
Comment (dot5enko) [reply app/components/Utility/CryptoLinkFormatter.jsx]: yes, this is handled in render() method.

File: app/components/Utility/CryptoLinkFormatter.jsx (hunk 3 above)
Comment (dot5enko) [reply app/components/Utility/CryptoLinkFormatter.jsx]: size is the only required prop. If i provide default values to another props like message we would get non useful info in link, like next: ethereum:0xded7ad236aaf95465c47?message= 
or double check if it's not empty + "noneptiness" cond for each type :)

File: app/components/Utility/CryptoLinkFormatter.jsx (hunk 4 above)
Comment (dot5enko) [reply app/components/Utility/CryptoLinkFormatter.jsx]: that was a typo, thanks - fixed it.  If some prop was not found - better to ommit whole qr creating then give a user wrong qr.

File: app/components/Utility/CryptoLinkFormatter.jsx (hunk 3 above)
Comment (sschiessl-bcp) [reply app/components/Utility/CryptoLinkFormatter.jsx]: This true, and maybe more personal preference.

Using "undefined" as a defined value is bad practice, IMO, but it's fine. undefined is no assigned value, whereas null is an assigned value.

**Comment by dot5enko** [comment]
@sschiessl-bcp done

**Pull request merged**